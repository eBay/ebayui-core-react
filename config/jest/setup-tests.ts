import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
