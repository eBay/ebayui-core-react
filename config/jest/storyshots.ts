import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
