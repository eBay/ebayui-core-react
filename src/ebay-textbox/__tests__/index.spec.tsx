import React from 'react'
import { fireEvent, render } from '@testing-library/react';
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayTextbox, EbayTextboxPostfixIcon } from '../index'

describe('<EbayTextbox>', () => {
    describe('on textbox change', () => {
        it('should fire an event', () => {
            const spy = jest.fn()
            const wrapper = render(
                <EbayTextbox onChange={spy} />
            )
            fireEvent.change(wrapper.getByRole('textbox'), { target: { value: `It's a spy!` } })

            expect(spy).toBeCalled()
        })
    })

    describe('on textbox button click', () => {
        it('textbox should fire an event', () => {
            const spy = jest.fn()
            const wrapper = render(
                <EbayTextbox onButtonClick={spy}>
                    <EbayTextboxPostfixIcon name="clear" buttonAriaLabel="Clear!"/>
                </EbayTextbox>
            )
            fireEvent.click(wrapper.getByRole('button'))

            expect(spy).toBeCalled()
        })

        it('button should fire an event', () => {
            const spy = jest.fn()
            const wrapper = render(
                <EbayTextbox>
                    <EbayTextboxPostfixIcon name="clear" buttonAriaLabel="Clear!" onClick={spy} />
                </EbayTextbox>
            )
            fireEvent.click(wrapper.getByRole('button'))

            expect(spy).toBeCalled()
        })
    })

    describe('on passing a ref', () => {
        it('should set the ref as the input reference', () => {
            const ref = React.createRef() as any
            render(<EbayTextbox ref={ref} />)

            expect(ref.current.tagName).toBe('INPUT')
        })
    })
})

initStoryshots({
    config: ({ configure }) =>
        configure(() => {
            require('./index.stories')
        }, module)
})
