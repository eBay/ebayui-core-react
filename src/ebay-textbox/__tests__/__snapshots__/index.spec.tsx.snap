// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-textbox Autofocused textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      class="textbox__control"
      placeholder="Should focus here"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Control value from outside 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      class="textbox__control"
      placeholder="Max 10 chars"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Default textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      class="textbox__control"
      type="text"
      value="EbayTextbox"
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Disabled textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      class="textbox__control"
      disabled=""
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Floating label 1`] = `
<DocumentFragment>
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label"
    >
      Floating label
    </label>
    <span
      class="textbox"
    >
      <input
        class="textbox__control"
        type="text"
        value=""
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Floating label invalid 1`] = `
<DocumentFragment>
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label floating-label__label--invalid"
    >
      Invalid Floating label
    </label>
    <span
      class="textbox"
    >
      <input
        aria-invalid="true"
        class="textbox__control"
        type="text"
        value=""
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Floating label with autofocus 1`] = `
<DocumentFragment>
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label floating-label__label--animate"
    >
      Autofocused field
    </label>
    <span
      class="textbox"
    >
      <input
        class="textbox__control"
        type="text"
        value=""
      />
    </span>
  </span>
   
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label"
    >
      Regular field
    </label>
    <span
      class="textbox"
    >
      <input
        class="textbox__control"
        type="text"
        value=""
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Floating label with placeholder, controlled 1`] = `
<DocumentFragment>
  <p>
    <span
      class="floating-label"
    >
      <label
        class="floating-label__label"
      >
        Will convert to lowercase
      </label>
      <span
        class="textbox"
      >
        <input
          class="textbox__control"
          placeholder="Enter some UPPERCASE"
          size="30"
          type="text"
          value=""
        />
      </span>
    </span>
  </p>
  <p>
    <button
      class="btn btn--secondary"
    >
      Change text
    </button>
  </p>
  <p>
    <button
      class="btn btn--secondary"
    >
      Clear
    </button>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Floating label with value 1`] = `
<DocumentFragment>
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label"
    >
      Floating label
    </label>
    <span
      class="textbox"
    >
      <input
        class="textbox__control"
        type="text"
        value="Default value"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Fluid textbox 1`] = `
<DocumentFragment>
  <div
    class="textbox"
  >
    <input
      class="textbox__control textbox__control--fluid"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Invalid textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      aria-invalid="true"
      class="textbox__control"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Large textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      class="textbox__control textbox__control--large"
      placeholder="placeholder text"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Multiline invalid textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <textarea
      aria-invalid="true"
      class="textbox__control"
      type="text"
    >
      some default value
    </textarea>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Multiline textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <textarea
      class="textbox__control"
      type="text"
    >
      some default value
next line
    </textarea>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Password textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      class="textbox__control"
      type="password"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Placeholder textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      class="textbox__control"
      placeholder="placeholder text"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox Ref forwarding 1`] = `
<DocumentFragment>
  <span
    class="textbox"
  >
    <input
      class="textbox__control"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-textbox With icon 1`] = `
<DocumentFragment>
  <div>
    <p>
      <span
        class="textbox"
      >
        <svg
          aria-hidden="true"
          class="icon icon--messages"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-messages"
          />
        </svg>
        <input
          class="textbox__control"
          placeholder="email"
          type="text"
          value=""
        />
      </span>
    </p>
    <p>
      <span
        class="textbox textbox--icon-end"
      >
        <input
          class="textbox__control"
          placeholder="username"
          type="text"
          value=""
        />
        <svg
          aria-hidden="true"
          class="icon icon--user-profile"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-user-profile"
          />
        </svg>
      </span>
    </p>
    <p>
      <span
        class="textbox textbox--icon-end"
      >
        <svg
          aria-hidden="true"
          class="icon icon--search"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-search"
          />
        </svg>
        <input
          class="textbox__control"
          placeholder="search"
          type="text"
          value=""
        />
        <button
          aria-label="Clear"
          class="icon-btn icon-btn--transparent"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--clear"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-clear"
            />
          </svg>
        </button>
      </span>
    </p>
  </div>
</DocumentFragment>
`;
